### DISCUSSION TITLE
Deprecate old stuff

### DISCUSSION THREAD

**Discussion by sandstrom** [libvips/libvips#2883] [General]
Libvips is an awesome project! 🏅 and one with a long life-span. Any long-lived software project, however, will accumulate some legacy code over time.

I just wanted to open this issue to encourage deprecation of things that aren't used much, or has been superseded by newer things. For example, image operations that are rarely used or image formats that have little use 2022.

Deprecating old code makes it easier to maintain the project over time.

**Comment by jcupitt** [comment]
Sure, we have some deprecation planned:

- The autotools build system will be going during 8.14, hopefully.
- I'd like to get rid of the tiny and not-very-useful scripts we ship for historical reasons (`batch_image_convert` etc.)
- The `benchmark` thing should be in a separate repo
- We build with deprecated components enabled. These could possibly be off by default.

**Reply by jcupitt** [reply]
.... other suggestions very welcome, of course

**Reply by sandstrom** [reply]
Sounds great! 🏅 

I've mentioned translations before, but I remember that you wanted to keep them. No other suggestions right now, I don't know the internals of libvips very well.

A big fan of the library though! 🎉

**Comment by sandstrom** [comment]
Just wanted to chime in here again. 

I love libvips, but I think the project would be easier to maintain if it would deprecate old stuff a bit more aggressively. 

Here are a few more suggestions:

- ORC SIMD backend (Highway seems to be the successor?)
- vips7 compat layer
- V4L1 video capture (I think it was deprecated in Linux kernel 2010?)
- All translations [as outlined here](https://github.com/libvips/libvips/discussions/2079)
- Analyze 7.5 format (I think NIfTI-1/NIfTI-2 are the popular successors)
- Matlab format loader (foreign/matlab.c) since it seems niche (Matlab can export PNG/TIFF)

These are just guesses from me, you probably have many more ideas on things that could be deprecated.

**Reply by jcupitt** [reply]
Hi again, good news! Git master (and 8.19) have `-Ddeprecated=false` (don't build deprecated components) as the default, so quite a few things are going very soon.

This is because the last significant vips7 application has now finally been replaced:

https://github.com/libvips/nip4

There's more to do and your ideas are good ones. We should probably think about better names for the deprecation options.

**Reply by sandstrom** [reply]
Sounds good! 

I think it's mostly down to Vips beings such a successful project, that's been around for long. 

But that also means legacy stuff will eventually accumulate. And pruning that off with some regularity will make it easier to evolve the project and focus on the core. 

Not building deprecated code is good, but I would also just drop  deprecated code from the code base in the familiar way: deprecate in one major, drop in the next. And don't be afraid of releasing a new major at least once a year. 

The few that need it can continue on the older release. 

Again, huge fan of the project and your amazing work! 🙂

**Reply by kleisauke** [reply]
> * ORC SIMD backend (Highway seems to be the successor?)

Example commit here: https://github.com/kleisauke/libvips/commit/adabe8e9b5e3fc8503fec952be340ae560c0753a.

However, I think it is still too early to remove support for this. Neither RHEL 8 nor Debian 12 ships Highway >= 1.0.5 in their package managers, see e.g.:
https://github.com/libvips/libvips/discussions/3886#discussioncomment-8736014

and:
[![highway-devel - EPEL 8](https://repology.org/badge/version-for-repo/epel_8/highway-simd-library.svg?header=highway-devel%20-%20EPEL%208)](https://packages.fedoraproject.org/pkgs/highway/highway-devel/epel-8.html)

For similar reasons, we also cannot require [require Meson >= 0.60](https://github.com/libvips/libvips/compare/HEAD...kleisauke:meson-0.60). We need to wait until [31 May 2029](https://endoflife.date/rhel) before doing this. :(
